packages/lesswrong/server/repos/PostViewTimesRepo.ts

```diff
@@ -49,20 +49,57 @@ class PostViewTimesRepo extends IncrementalViewRepo<"PostViewTimes"> {
         post_id,
         date_trunc('day', timestamp)
     `,
-      [windowStart, windowEnd]
+      [startDateFormatted, endDateFormatted]
     );
 
-    // Manually convert the number fields from string to number (as this isn't handled automatically)
+    // Manually convert:
+    // - The number fields from string to number (as this isn't handled automatically)
+    // - The date fields to UTC strings (NOTE: this actually changes the time represented, because in the conversion
+    //   from postgres to JS the timezone of the server is applied to the raw date, so 2023-04-01 00:00:00 becomes e.g.
+    //   2023-04-01 00:00:00+0100 (BST), which is wrong)
     const typedResults = results.map((views) => ({
       ...views,
+      windowStart: moment(views.windowStart)
+        .add(moment(views.windowStart).utcOffset(), "minutes")
+        .utc()
+        .format("YYYY-MM-DD HH:mm:ss"),
+      windowEnd: moment(views.windowEnd)
+        .add(moment(views.windowEnd).utcOffset(), "minutes")
+        .utc()
+        .format("YYYY-MM-DD HH:mm:ss"),
       viewCount: parseInt(views.viewCount),
       uniqueViewCount: parseInt(views.uniqueViewCount),
     }));
 
     return typedResults;
   }
 
-  async upsertData({ data }: { data: Omit<DbPostViewTime, "_id" | "schemaVersion" | "createdAt" | "legacyData">[] }) {
+  /**
+   * Delete rows where there is any overlap with the given (startDate, endDate) range
+   */
+  async deleteRange({
+    startDate,
+    endDate,
+  }: {
+    startDate: Date;
+    endDate: Date;
+  }): Promise<void> {
+    // Round startDate (endDate) down (up) to the nearest UTC date boundary
+    const startDateFormatted = moment.utc(startDate).startOf('day').format('YYYY-MM-DD HH:mm:ss');
+    const endDateFormatted = moment.utc(endDate).endOf('day').format('YYYY-MM-DD HH:mm:ss');
+
+    await this.getRawDb().none(
+      `
+      DELETE FROM "PostViewTimes"
+      WHERE
+        ("windowEnd" > $1 AND "windowEnd" < $2)
+        OR ("windowStart" > $1 AND "windowStart" < $2)
+      `,
+      [startDateFormatted, endDateFormatted]
+    );
```

Comment:
Because the incorrect rows are offset by an hour they won't be overwritten by upserting, so we need to additionally delete them